src/foam/core/stdlib.js

```diff
@@ -586,10 +586,8 @@ foam.LIB({
     function getTime(d) { return ! d ? 0 : d.getTime ? d.getTime() : d ; },
     function equals(a, b) { return this.getTime(a) === this.getTime(b); },
     function compare(a, b) {
-      if ( ! foam.Date.isInstance(b) && ! foam.Number.isInstance(b) ) return 1;
-      a = this.getTime(a);
-      b = foam.Number.isInstance(b) ? b : this.getTime(b);
-      return foam.Number.compare(a, b);
+      if ( ! foam.Date.isInstance(b) ) return 1;
```

Comment:
That would blow up the stack if this predicate is true and b is of lower precedence than Date, I think.